services/QuillLMS/spec/models/schools_users_spec.rb

```diff
@@ -36,13 +36,15 @@
   end
 
   describe '#reset_teacher_verification' do
+    let(:other_school) { create(:school) }
+
     context 'when user has teacher_info' do
       let(:teacher_info) { create(:teacher_info, :verification_approved, verifier_id: 123) }
       let(:user_with_teacher_info) { create(:user, teacher_info:) }
-      let(:schools_user) { create(:schools_users, user: user_with_teacher_info, school:) }
+      let!(:schools_user) { create(:schools_users, user: user_with_teacher_info, school:) }
 
-      it 'resets teacher_info verification status and verifier_id after save' do
-        schools_user.save
+      it 'resets teacher_info verification status and verifier_id after school is changed' do
+        schools_user.update(school: other_school)
```

Comment:
The relevant part of this test isn't part of this PR, so definitely take or leave this, but again I'd do this with
```ruby
expect { schools_user.update(school: other_school) }
  .to change(teacher_info, :verification_status).to(nil)
  .and change(teacher_info, :verifier_id).to(nil)
```